posts/2021-06-11-request-timing-21006.adoc

```diff
@@ -0,0 +1,193 @@
++
+To fix this issue it was decided to change the default of the `maxKeepAliveRequests` option to `-1`. 
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/16743[Pull in MyFaces 2.3.9]
++
+The `jsf-2.3` feature in Open Liberty makes use of link:https://myfaces.apache.org/#/core23[Apache MyFaces] JavaServer Faces implementation. With the release of Apache MyFaces 2.3.9 a number of improvements and bug fixes have been made, for more information visit the link:https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=10600&version=12349634[Apache MyFaces 2.3.9 changelog].
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/15023[WASReqURLOidc cookie encodes the request url but doesn't decoded it upon successful redirection]
++
+When calling an OIDC-protected URL that includes query parameters, the URL in the `WASReqURLOidc` cookie was incorrectly encoded. This would cause the URL query parameters to not be decoded correctly upon successful redirection to the original request URL. To fix this issue, instead of encoding the query string in the original request URL, we are leaving it as-is. That should ensure we ultimately use the correct original value of the query string. During the creation of the `WASReqURLOidc` cookie, the cookie value is encoded to mitigate against malformed and potentially malicious cookie values anyway. The `OidcRedirectServlet` class is also updated to use slightly different code to read the cookie so that the appropriate respective decoding is done.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/12778[mpJWT-1.1 configured by using jwksUri results in CWWKS5523E at the first jwt token presented to the server]
++
+The `JwkRetriever` class contained a bug that caused it to potentially return an incorrect JWK from an internal cache. The wrong key could be returned if a JWK document containing a single JWK was previously fetched and cached, followed by the Liberty server receiving a JWT with a `kid` value that didn’t match the cached JWK. Instead of attempting to re-fetch the JWK document to find a matching JWK, the Liberty runtime would return the cached JWK as the key to use to verify the JWT’s signature. That would cause a JWT signature verification error, which would fail the request.
+The `JwkRetriever` class is now updated to return the sole cached JWK only if the JWT does not contain a key identifier (e.g. `kid`, `x5t`, or `use` values). If the JWT does contain one of those values, but a key cannot be found in the cache that matches one of those values, the runtime will now re-fetch the JWK document and see if it can find a matching key from there.
+
+== Known issues
+
+
```

Comment:
* link:https://github.com/OpenLiberty/open-liberty/issues/17465[Remove system out] An extra message is output to the console during server startup.  It will only show when creating a certificate and on the first server start.  Subsequent ones will not see it.  
* link:https://github.com/OpenLiberty/open-liberty/issues/17478[Invalid featureUtility help parameters shown] Invalid command line optional parameters (--featuresBom) and (--to) are shown with "featureUtility help installFeature". Invalid command line optional parameter (--featuresBom) is shown with "featureUtility help installServerFeatures"

Reply:
I also added a small amount of text below the title to give a slight explanation. Please review and let me know if you prefer to remove overall or change it. 

Texted added into the section:
"In this release there is a couple of issues that have already been detected. If you would like to find out more you can check out the corresponding Github Issues linked bellow."